spec.html

```diff
@@ -507,38 +511,32 @@ <h1>InnerModuleEvaluation( _module_, _stack_, _index_ )</h1>
       </emu-note>
     </emu-clause>
 
-    <emu-clause id="sec-execute-cyclic-module" aoid="ExecuteCyclicModule">
-      <h1><ins>ExecuteCyclicModule ( _module_ )</ins></h1>
+    <emu-clause id="sec-execute-async-module" aoid="ExecuteAsyncModule">
+      <h1><ins>ExecuteAsyncModule ( _module_ )</ins></h1>
       <emu-alg>
-        1. Assert: _module_.[[Status]] is `"evaluating"` or `"evaluating-async"`.
-        1. If _module_.[[Async]] is *false*, then:
-          1. Let _result_ be _module_.ExecuteModule().
-          1. If _result_ is a normal completion,
-            1. Perform ! CyclicModuleExecutionFulfilled(_module_).
-          1. Otherwise,
-            1. Perform ! CyclicModuleExecutionRejected(_module_, _result_.[[Value]]).
-        1. Otherwise,
-          1. Let _capability_ be ! NewPromiseCapability(%Promise%).
-          1. Let _stepsFulfilled_ be the steps of a CallCyclicModuleFulfilled function as specified below.
-          1. Let _onFulfilled_ be CreateBuiltinFunction(_stepsFulfilled_, &laquo; [[Module]] &raquo;).
-          1. Set _onFulfilled_.[[Module]] to _module_.
-          1. Let _stepsFulfilled_ be the steps of a CallCyclicModuleRejected function as specified below.
-          1. Let _onRejected_ be CreateBuiltinFunction(_stepsRejected_, &laquo; [[Module]] &raquo;).
-          1. Set _onRejected_.[[Module]] to _module_.
-          1. Perform ! PerformPromiseThen(_capability_.[[Promise]], _onFulfilled_, _onRejected_).
-          1. Perform ! _module_.ExecuteModule(_capability_).
-          1. Return.
+        1. Assert: _module_.[[Status]] is `"evaluating-async"`.
+        1. Assert: _module_.[[Async]] is *true*.
+        1. Let _capability_ be ! NewPromiseCapability(%Promise%).
+        1. Let _stepsFulfilled_ be the steps of a CallAsyncModuleFulfilled function as specified below.
+        1. Let _onFulfilled_ be CreateBuiltinFunction(_stepsFulfilled_, &laquo; [[Module]] &raquo;).
+        1. Set _onFulfilled_.[[Module]] to _module_.
+        1. Let _stepsRejected_ be the steps of a CallAsyncModuleRejected function as specified below.
+        1. Let _onRejected_ be CreateBuiltinFunction(_stepsRejected_, &laquo; [[Module]] &raquo;).
+        1. Set _onRejected_.[[Module]] to _module_.
+        1. Perform ! PerformPromiseThen(_capability_.[[Promise]], _onFulfilled_, _onRejected_).
+        1. Perform ! _module_.ExecuteModule(_capability_).
+        1. Return.
       </emu-alg>
 
-      <p>A CallCyclicModuleRejected function is an anonymous built-in function with a [[Module]] internal slot. When a CallCyclicModuleRejeced function is called that expects no arguments it performs the following steps:</p>
+      <p>A CallAsyncModuleRejected function is an anonymous built-in function with a [[Module]] internal slot. When a CallAsyncModuleRejeced function is called that expects no arguments it performs the following steps:</p>
```

Comment:
There's significant confusion here which of these handles fulfillment and which handles rejection.